assembl/lib/rds_token_url.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime, timedelta
+
+from sqlalchemy.engine.url import URL
+import boto3
+
+
+class RdsTokenUrl(URL):
+    """An URL that uses IAM connection tokens as passwords"""
+    def __init__(
+            self, drivername, region=None, username=None, host=None,
+            session_name=None, port=None, database=None, query=None):
+        super(RdsTokenUrl, self).__init__(
+            drivername, username=username, host=host, port=port or 5432,
+            database=database, query=query)
+        self.aws_region = region
+
+    def get_rds_client(self):
+        if not getattr(self, 'rds_client', None):
+            self.rds_client = boto3.client(
+                'rds', self.aws_region)
+        return self.rds_client
+
+    @property
+    def password(self):
+        rds = self.get_rds_client()
+        return rds.generate_db_auth_token(
+            DBHostname=self.host,
+            Port=self.port,
+            DBUsername=self.username,
+            Region=self.aws_region)
+
+
+class IamRoleRdsTokenUrl(RdsTokenUrl):
+    """Assume a specific role before you get the client.
+    AWS bug: you have to assume your own role to connect."""
+    def __init__(
+            self, drivername, iam_role, region=None, username=None,
+            host=None, port=None, database=None,
+            query=None, session_name=None):
+        super(IamRoleRdsTokenUrl, self).__init__(
+            drivername, region=region, username=username, host=host,
+            port=port, database=database, query=query)
+        self.iam_role = iam_role
+        self.sts_session_name = session_name or 'assembl'
+        self.sts_client = boto3.client('sts', self.aws_region)
+
+    def _needs_sts_role(self):
+        role = getattr(self, 'sts_role', None)
+        if role:
+            expiry = role['Credentials']['Expiration']
+            return (datetime.now() - expiry) < timedelta(minutes=10)
```

Comment:
Should `return False`, in case role is not found as an attribute (Explicit is better than implicit)